test/data/mini_example/mini_example.sam

```diff
@@ -0,0 +1,76 @@
+read033	2048	chr1	181	60	8M42S	*	0	0	ATATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAATA	*	AS:i:8	NM:i:0
+read033	256	chr1	181	60	8M8I34M	*	0	0	ATATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAATA	*	AS:i:38	NM:i:8
+read034	0	chr1	181	60	7S43M	*	0	0	TATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAG	*	AS:i:43	NM:i:0	SA:Z:chr1,182,+,7M43S,60,7,0;
+read035	0	chr1	181	60	1S49M	*	0	0	TATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAGGTCTCG	*	AS:i:49	NM:i:0
+read036	0	chr1	181	60	50M	*	0	0	ATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAGGTCTCGG	*	AS:i:50	NM:i:0
+read034	256	chr1	182	60	7M8I35M	*	0	0	TATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAG	*	AS:i:38	NM:i:8
+read034	2048	chr1	182	60	7M43S	*	0	0	TATATTTATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAG	*	AS:i:7	NM:i:0
+read035	256	chr1	189	60	9S41M	*	0	0	TATATATTTTACGGGATATAACGATCGGATCGGATCGATAATAGGTCTCG	*	AS:i:41	NM:i:0
+read037	0	chr1	222	60	45M20D5M	*	0	0	AGGTCTCGGTTGCCAACTGATCGTACCAAATATTTCTGCGGGGCTTCGGC	*	AS:i:46	NM:i:20
+read038	0	chr1	223	60	44M20D6M	*	0	0	GGTCTCGGTTGCCAACTGATCGTACCAAATATTTCTGCGGGGCTTCGGCT	*	AS:i:46	NM:i:20
+read039	0	chr1	251	60	16M17I16M1S	*	0	0	ATATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTAC	*	AS:i:27	NM:i:17
+read039	256	chr1	251	60	16M20D13M21S	*	0	0	ATATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTAC	*	AS:i:24	NM:i:20
+read039	256	chr1	251	60	16M34S	*	0	0	ATATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTAC	*	AS:i:16	NM:i:0	SA:Z:chr1,287,+,16S13M21S,60,13,0;chr1,283,+,29S4M17S,60,4,0;chr1,267,+,33S16M1S,60,16,0;
+read040	0	chr1	252	60	15M17I16M17D2M	*	0	0	TATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTACA	*	AS:i:24	NM:i:34
+read040	256	chr1	252	60	15M20D13M20I2M	*	0	0	TATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTACA	*	AS:i:20	NM:i:40
+read040	256	chr1	252	60	15M35S	*	0	0	TATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTACA	*	AS:i:15	NM:i:0	SA:Z:chr1,287,+,15S13M22S,60,13,0;chr1,283,+,28S4M18S,60,4,0;chr1,267,+,32S16M2S,60,16,0;
+read039	2305	chr1	267	60	33S16M1S	*	0	0	ATATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTAC	*	AS:i:16	NM:i:0
+read040	2305	chr1	267	60	32S16M2S	*	0	0	TATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTACA	*	AS:i:16	NM:i:0
+read041	0	chr1	267	60	11S16M17D23M	*	0	0	AACGGTTAGAGCGCCCCTCCGCGATTACACCCATGCGGATTATAAACGGG	*	AS:i:34	NM:i:20	SA:Z:chr1,283,+,7S4M39S,60,4,0;chr1,293,+,7M20I23M,60,25,20;
+read039	2305	chr1	283	60	29S4M17S	*	0	0	ATATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTAC	*	AS:i:4	NM:i:0
+read040	2305	chr1	283	60	28S4M18S	*	0	0	TATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTACA	*	AS:i:4	NM:i:0
+read041	2048	chr1	283	60	7S4M39S	*	0	0	AACGGTTAGAGCGCCCCTCCGCGATTACACCCATGCGGATTATAAACGGG	*	AS:i:4	NM:i:0
+read039	2305	chr1	287	60	16S13M21S	*	0	0	ATATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTAC	*	AS:i:13	NM:i:0
+read040	2305	chr1	287	60	15S13M22S	*	0	0	TATTTCTGCGGGGCTTCGGCTAACGGTTAGAGCGCCCCTCCGCGATTACA	*	AS:i:13	NM:i:0
+read041	2048	chr1	293	60	7M20I23M	*	0	0	AACGGTTAGAGCGCCCCTCCGCGATTACACCCATGCGGATTATAAACGGG	*	AS:i:25	NM:i:20
```

Comment:
The last 23 bases of this read have already been aligned in the alignment at position 267 (11S16M17D23M). When you change the CIGAR string from 7M20I23M to 7M43M you get a nice split of the read into three non-overlapping segments at positions 267, 283 and 293.